ndb_api/db_request_types.go

```diff
@@ -29,12 +29,13 @@ type DatabaseProvisionRequest struct {
 	CreateDbServer           bool             `json:"createDbserver"`
 	NodeCount                int              `json:"nodeCount"`
 	NxClusterId              string           `json:"nxClusterId"`
-	SSHPublicKey             string           `json:"sshPublicKey"`
+	SSHPublicKey             string           `json:"sshPublicKey,omitempty"`
```

Comment:
Sorry can you educate me on this one? Doesn't "omitempty" cause the generated CRD to mark this field as optional? That is not an optional field in our APIs right?

Reply:
The SSHKey field is not required for MSSQL Databases. That is why I marked it as omitempty. There is engine specific validation for this field in the controller that would error out if SSHKey is not present for other engines